functions/index.js

```diff
@@ -0,0 +1,58 @@
+// we need ES5 requires becauses this folder does not have babel configured
+
+const admin = require('firebase-admin')
+const functions = require('firebase-functions')
+
+exports.onDiagnoseCreationAddTimestamp = functions.firestore
+  .document('diagnoses/{diagnoseId}')
+  .onCreate((snapshot, context) => {
+    const timestamp = admin.firestore.FieldValue.serverTimestamp()
+    return snapshot.ref.update({
+      createdAt: timestamp,
+      updatedAt: timestamp
+    })
+  })
+
+exports.onDiagnoseUpdateModifyTimestamp = functions.firestore
+  .document('diagnoses/{diagnoseId}')
+  .onUpdate((snapshot, context) => {
+    const currentTimestamp = admin.firestore.Timestamp.now()
+    const previousTimestamp = snapshot.before.data().updatedAt
+    const FIVE_MINUTES_IN_MILLISECONDS = 300000
+    console.log(`previousTimeStamp: ${previousTimestamp.toMillis()}  currentTimeStamp: ${currentTimestamp.toMillis()}`)
+
+    if (Math.abs(previousTimestamp.toMillis() - currentTimestamp.toMillis()) > FIVE_MINUTES_IN_MILLISECONDS) {
+      return snapshot.after.ref.update({
+        updatedAt: currentTimestamp
+      })
+    }
+
+    return null
+  })
+
+exports.onUserCreationAddTimestamp = functions.firestore
```

Comment:
We can refactor this and create a single function to do this. Something as `setTimestampsToRecord('users/{userId}')`

Reply:
cc @nahudalla